syntax = "proto3";

package replication;

service ReplicationService {
  rpc ReplicateTopic (TopicMessage) returns (Ack);
  rpc ReplicateQueueMessage (QueueMessage) returns (Ack);
}

message TopicMessage {
  string topic = 1;
  string message = 2;
}

message QueueMessage {
  string queue = 1;
  string message = 2;
}

message Ack {
  bool success = 1;
}
